modules/data.remote/R/soilgrids.R

```diff
@@ -0,0 +1,224 @@
+##' @title soilgrids.soc.extract
+##' @name  soilgrids.soc.extract
+##' 
+##' @param site_info A BETYdb site info dataframe containing at least each site ID, sitename, 
+##' latitude, longitude, and time_zone. e.g. c(site_qry$id, site_qry$sitename, site_qry$lon,
+##' site_qry$lat, site_qry$time_zone)
+##' @param outdir Directory to store SOC data.  Format *.Rdata
+##' @param verbose logical. If TRUE show the extraction progress bar
+##' 
+##' ##' @examples
+##' \dontrun{
+##' 
+##' # Example 1 - using the modex.bnl.gov BETYdb and site IDs to extract data
+##' db <- 'betydb'
+##' host_db <- 'modex.bnl.gov'
+##' db_port <- '5432'
+##' db_user <- 'bety'
+##' db_password <- 'bety'
+##' 
+##' bety <- list(user='bety', password='bety', host='modex.bnl.gov',
+##' dbname='betydb', driver=RPostgres::Postgres(),write=FALSE)
+##' 
+##' con <- DBI::dbConnect(drv=bety$driver, dbname=bety$dbname, host=bety$host, 
+##' password=bety$password, user=bety$user)
+##' 
+##' suppressWarnings(site_qry <- glue::glue_sql("SELECT *, ST_X(ST_CENTROID(geometry)) AS lon,
+##' ST_Y(ST_CENTROID(geometry)) AS lat FROM sites WHERE id IN ({ids*})",
+##' ids = c("676","622","678","766","764"), .con = con))
+##' 
+##' suppressWarnings(qry_results.1 <- DBI::dbSendQuery(con,site_qry))
+##' suppressWarnings(qry_results.2 <- DBI::dbFetch(qry_results.1))
+##' DBI::dbClearResult(qry_results.1)
+##' dbDisconnect(con)
+##' 
+##' site_info <- qry_results.2
+##' verbose <- TRUE
+##' system.time(result_soc <- soilgrids.soc.extract(site_info=site_info, verbose=verbose))
+##' result_soc
+##' 
+##' }
+##' 
+##' @importFrom reshape2 melt
+##' @importFrom terra vect project rast extract
+##' @importFrom stats qgamma optim
+##' @importFrom utils flush.console setTxtProgressBar txtProgressBar
+##' 
+##' @return list of two dataframes containing the mean SOC values per location and depth
+##' and the corresponding standard deviation values (uncertainties) for each location 
+##' and depth. Output column names are c("site_id","lat","lon","0-5cm","5-15cm","15-30cm",
+##' "30-60cm","60-100cm","100-200cm")
+##' 
+##' @export
+##' @author Qianyu Li, Shawn P. Serbin
+##' 
+soilgrids.soc.extract <- function (site_info=NULL, outdir=NULL, verbose=TRUE) {
+  
+  
+  if (is.null(site_info)) {
+    stop("Please provide a BETY DB site list containing at least the site id and PostGIS geometry\
+    as lon and lat")
+  }
+  
+  # prepare site info for extraction
+  internal_site_info <- data.frame(site_info$id, site_info$sitename, site_info$lat, 
```

Comment:
1. why redefine the site_info arg into a new internal variable instead of just using site_info?
2. If you do need to do this, why are you not defining variable names in your data frame? Current practice results in a lot of unnecessarily redundant variable names e.g. internal_site_info$site_info.lon instead of just site_info$lon

Reply:
No longer redefining in the function